src/daemon/collector.cc

```diff
@@ -11,15 +11,18 @@ namespace dist_clang {
 namespace daemon {
 
 Collector::Collector(const proto::Configuration& configuration)
-    : BaseDaemon(configuration), local_(configuration.collector().local()) {
+    : BaseDaemon(configuration) {
   CHECK(configuration.has_collector());
+  conf_.reset(new proto::Configuration(configuration));
```

Comment:
Since all descendants of `BaseDaemon` now have the `conf_` - let's put it in the base class and do this check in it's constructor.